File: 509-fibonacci-number/509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        int[] lastTwo =new int[2];
        lastTwo[0] = 0;
        lastTwo[1] = 1;
        int count =2;
        while(count <= n) {
            int nextNum = lastTwo[0] + lastTwo[1];
            lastTwo[0] = lastTwo[1];
            lastTwo[1] = nextNum;
            count++;
        }
        
        return n>0 ? lastTwo[1] : lastTwo[0];
        
    }
    
   
        
}

File: 509-fibonacci-number/509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        memorize = {0:0,1:1}
        return self.getNFib(n, memorize)
    
    def getNFib(self, n, memorize) :
        if n in memorize:
            return memorize[n]
        else:
            memorize[n]=self.getNFib(n-1, memorize) + self.getNFib(n-2, memorize)
            return memorize[n]
    
        
        
        